rustle_cli/src/main.rs

```diff
@@ -1,3 +1,281 @@
+use std::{fs::{self, ReadDir}, path::{Path, PathBuf}, io::ErrorKind, env::current_dir};
+
+
+use rustle::{compile_file_to_js, parse_file};
+use clap::{Parser, Subcommand};
+use console::style;
+
+
+#[derive(Parser)]
+#[command(name = "Rustle")]
+#[command(version = "0.0.1")]
+#[command(about = "Svelte compiler rewritten in Rust", long_about = None)]
+struct Cli {
+    file: Option<String>,
+
+    #[arg(short = 'o', long = "output")]
+    output: Option<String>,
+
+    #[arg(short = 'y', long = "yes")]
+    yes: Option<String>,
+
+    #[command(subcommand)]
+    command: Option<Sub>
+}
+
+
+#[derive(Subcommand)]
+enum Sub {
+    Parse {
+        file: Option<String>,
+
+        #[arg(short = 'p', long = "pretty", default_value_t=false)]
+        pretty: bool,
+
+        #[arg(short = 'o', long = "output")]
+        output: Option<String>
+    }
+}
+
+
 fn main() {
-    println!("Hello, world!");
+    let cli = Cli::parse();
+
+    if let Some(file) = cli.file.as_deref() {
+        let res = match fs::metadata(file) {
+            Ok(md) => {
+                if md.is_dir() {
+                    let files = match fs::read_dir(file) {
+
+                        Ok(rd) => {
+                            let mut output = current_dir().unwrap();
+                            output.push("./dist");
+                            let dir_creation_result = fs::create_dir(output.clone());
```

Comment:
try replacing this with a `match` statement

Reply:
Fixed.